test/run.js

```diff
@@ -0,0 +1,25 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+const { runCLI } = require('jest')
+const jestConfig = require('../jest.config.js')
+
+const sessionPlugins = ['fastify-secure-session', '@fastify/session']
+
+async function runTests(sessionPlugin) {
+  process.env.SESSION_PLUGIN = sessionPlugin
+
+  console.log('Running tests with session', sessionPlugin, '\n')
+  const result = await runCLI(jestConfig, ['./'])
+  if (result.results.success) {
+    console.log(`Tests completed\n`)
+  } else {
+    console.error(`Tests failed\n`)
+  }
+}
+
+async function start() {
+  for (const sessionPlugin of sessionPlugins) {
+    await runTests(sessionPlugin)
+  }
+}
+
+start().catch((err) => console.error(err))
```

Comment:
Do you think we could avoid these non-standard scripts to make it easy for new contributors to know how to run the tests? This kinda stuff is generally unexpected and weird such that most folks wouldn't guess to look for it, and it also couples to jests internals a bit. Jest test and tap tests are just global function calls -- so couldn't we just use a function to duplicate the test executions?

```typescript
const suite = (sessionPluginName: string) => {
  describe(`${sessionPluginName} tests`, () => {
    test("one thing", () => { ... })
    test("another thing", () => { ... })
  })
}

suite("@fastify/session");
suite("fastify-secure-session");
```

That way, all the nice jest features like test timing, the log output, regex filtering, etc all just work because the test cases are all normal test cases, not part of some outer layer of double execution with environment variables.

Reply:
I implemented the way you described it. Seems much cleaner now. Thanks!